nix/shared.nix

```
@@ -28,6 +28,7 @@ let
                 overrideCabal = pkgsNew.haskell.lib.overrideCabal;
                 optionalString = pkgsNew.stdenv.lib.optionalString;
                 isDarwin = pkgsNew.stdenv.isDarwin;
+                enableLibraryProfiling = true;
```

I don't think this has any effect. use this:
```suggestion
                enableLibraryProfiling = pkgsNew.stdenv.lib.enableLibraryProfiling;
```
(see https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/haskell-modules/lib.nix#L168) and then wrap the derivations you want to enable profiling for in that function like I did with `failOnAllWarnings`.

---

Thx! That helps a lot.